build.gradle

```diff
@@ -76,58 +80,68 @@ allprojects {
-        runtimeOnly("com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:${jacksonVersion}")
-        runtimeOnly("com.fasterxml.jackson.dataformat:jackson-dataformat-smile:${jacksonVersion}")
-        runtimeOnly("com.xtdb:xtdb-jdbc:${xtdbVersion}")
-        runtimeOnly("com.xtdb:xtdb-s3:${xtdbVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid CVE-2020-13956
-        runtimeOnly("org.apache.httpcomponents:httpclient:${httpclientVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid CVE-2021-21295
-        runtimeOnly("io.netty:netty-codec-http2:${nettyVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid CVE-2021-43797
-        runtimeOnly("io.netty:netty-codec-http:${nettyVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid CVE-2021-37136, CVE-2021-37137
-        runtimeOnly("io.netty:netty-codec:${nettyVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid CVE-2020-0026
-        runtimeOnly("io.netty:netty-handler:${nettyVersion}")
-        // Following dependency is to ensure we use updated version of dependent library to avoid sonatype-2012-0050
-        runtimeOnly("commons-codec:commons-codec:${commonscodecVersion}")
-        runtimeOnly("com.xtdb:xtdb-metrics:${xtdbVersion}")
-        // Dependencies provided by Egeria itself
-        implementation("org.odpi.egeria:audit-log-framework:${egeriaVersion}")
-        implementation("org.odpi.egeria:repository-services-apis:${egeriaVersion}")
-        implementation("org.odpi.egeria:open-connector-framework:${egeriaVersion}")
-        implementation("org.slf4j:slf4j-api:${slf4jVersion}")
-        // Dependencies only used for the build-time tests
-        implementation("ch.qos.logback:logback-classic:${logbackVersion}")
-        testImplementation("org.testng:testng:${testngVersion}") {
-            // Exclude snakeyaml, which has open CVEs and is unused
-            exclude group: "org.yaml", module: "snakeyaml"
-            exclude group: "org.apache.ant", module: "ant"
-        }
-        testImplementation("org.odpi.egeria:repository-services-implementation:${egeriaVersion}")
-        testImplementation("org.odpi.egeria:connector-configuration-factory:${egeriaVersion}") {
-            // Exclude the graph repository, as unused and will extend to many other dependencies
-            exclude group: "org.odpi.egeria", module: "graph-repository-connector"
-        }
-        testImplementation("org.odpi.egeria:open-metadata-types:${egeriaVersion}")
+        // This adds constraints (which can be overriden below) for all dependencies egeria uses
+        implementation platform("org.odpi.egeria:egeria:${egeriaVersion}")
```

Comment:
What does this mean when we built an uber jar for the connector — will this include _all_ of Egeria in the uber jar?

(Hopefully not? Would greatly increase the size of the connector jar + pull in all kinds of dependencies that would expand the security / CVE footprint — when probably 90%+ of these dependencies are not actually used by the connector...)

Reply:
No - we will not pull in extra egeria dependencies beyond those explicitly added as dependencies in the gradle file.

I've used the build health plugin to try and optimize the dependencies

 the 'platform' specification is similar to 'import' in maven - it is just pulling in *dependency constraints*.
These are populated into a 'bom' in the egeria project -- I've just refactored that (the gradle version was wrong).

So where the same artifact is used in the connector, it should use the version in the constraint - aligning versions is important as the connector is in the same classpath as the egeria platform

So should be fine, but more testing and closer inspection to check.

This is a first pass & we may need to refine.

Reply:
For example:
```
ar -tf egeria-connector-xtdb-4.0-SNAPSHOT-jar-with-dependencies.jar| grep odpi | grep -v xtdb
org/odpi/
org/odpi/egeria/
org/odpi/egeria/connectors/
org/odpi/egeria/connectors/juxt/
```
But it does contain all the xtdb , clojure classes etc